openhtf/__init__.py

```diff
@@ -162,16 +164,26 @@ def plug_type_map(self):
     return plug_type_map
 
 
+def CreateArgParser():
+  """Creates an argparse.ArgumentParser for parsing command line flags."""
+  parser = argparse.ArgumentParser('OpenHTF-based testing', parents=[
+      conf.ARG_PARSER, user_input.ARG_PARSER, phase_executor.ARG_PARSER,
+      logs.ARG_PARSER])
+  return parser
+
+
 @functions.RunOnce
-def SetupFramework():
-  """Sets up various bits of the framework. Only needs to be called once."""
-  try:
-    sys.argv = FLAGS(sys.argv)  # parse flags
-  except gflags.FlagsError, e:  # pylint: disable=invalid-name
-    print '%s\nUsage: %s ARGS\n%s' % (e, sys.argv[0], FLAGS)
-    sys.exit(1)
-
-  logs.setup_logger()
+def SetupFramework(arguments=None):
+  """Sets up various bits of the framework. Only needs to be called once.
+
+  Args:
+    arguments: An argparse.Namespace-like object with at least the attributes
+        from logs.ARG_PARSER.
+  """
+  arguments = arguments or CreateArgParser().parse_args()
```

Comment:
it's a little weird that some modules use the action to save the flag values locally, but `logs.py` has to have the argument values pulled out of the namespace here and passed in.